Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/GUIMainWindow.java

```diff
@@ -0,0 +1,203 @@
+		statsPanel.setComponentOrientation(ComponentOrientation.LEFT_TO_RIGHT);
+		statsPanel.setLayout(null);
+		frame.getContentPane().add(statsPanel);
+
+		JLabel lblPlayerStats = new JLabel("Player Stats:");
+		lblPlayerStats.setBounds(29, 12, 195, 15);
+		statsPanel.add(lblPlayerStats);
+
+		//JLabel lblName = new JLabel(this.playerName());
+		lblName = new JLabel ("PlayerName");
+		lblName.setBounds(29, 39, 195, 15);
+		statsPanel.add(lblName);
+
+		//JLabel lblGamesWon = new JLabel("GamesWon");
+		lblGamesWon = new JLabel("GamesWon/GamesLost");
+		lblGamesWon.setBounds(29, 66, 195, 15);
+		statsPanel.add(lblGamesWon);
+
+		JLabel lblWrongLetters = new JLabel("Wrong Letters");
+		lblWrongLetters.setBounds(29, 148, 169, 15);
+		statsPanel.add(lblWrongLetters);
+
+		lblLettersTried = new JLabel("<Letters Here>");
+		lblLettersTried.setText("<html><p style=\"width:120px\">"+"a, b, c, d, e, f, g, h, i, j, k, l, m, n"+"</p></html>");
+		lblLettersTried.setBounds(29, 175, 169, 60);
+		statsPanel.add(lblLettersTried);
+
+		openedWord = new JLabel("- - - - - - ");
+		openedWord.setBounds(29, 121, 182, 15);
+		statsPanel.add(openedWord);
+
+		lblStatus = new JLabel("<Status>");
+		lblStatus.setFont(new Font("Courier New", Font.PLAIN, 12));
+		lblStatus.setBounds(12, 304, 212, 15);
+		statsPanel.add(lblStatus);
+		
+		separator = new JSeparator();
+		separator.setBounds(29, 96, 195, 13);
+		statsPanel.add(separator);
+
+		hangmanPanel = new GUIGamePanel();
+		frame.getContentPane().add(hangmanPanel);
+	}
+
+	@Override
+	public String playerName() {
+		String s = JOptionPane.showInputDialog(frame,"Enter a Player's Name. If one exists, will be loaded from file.","Player Selection/Creation",	JOptionPane.PLAIN_MESSAGE);
+		return s;
+
+	}
+
+	@Override
+	public void status() {
+		lblLettersTried.setText(game.getGuessedLetters().toString());
+		openedWord.setText(game.getOpened());
+		hangmanPanel.setHangmanStage(game.getHangmanStage());
+		hangmanPanel.repaint();
+		return;
+		//Useless in terms of GUI;
+
+	}
+
+	@Override
+	public boolean continuePrompt() {
+		if(firstRun){
+			firstRun = false;
+			this.frame.setVisible(true);
+			lblName.setText(game.getPlayer().getName());
+			lblGamesWon.setText(game.getPlayer().getScore().getGamesWon()+"/"+game.getPlayer().getScore().getGamesPlayed());
+			lblStatus.setText("Welcome "+game.getPlayer().getName());
+			return true;
+		}else{
+		return JOptionPane.showConfirmDialog(frame,"You've "+(game.hasWon()?"won.":"lost.")+"\nThe Word was "+game.getRightWord()+ "\nWould you Like to Quit?")==1;
+		//TODO fix integer Assignments
+		}
+	}
+
+	@Override
+	public char guessPrompt() {
+		String keyStroke=hangmanPanel.getPressedKey();
+		if(keyStroke==null){
+			return 0;
+		}else{
+			keyStroke=keyStroke.toLowerCase().trim();
+		}
+		return keyStroke.toCharArray()[0];
+	}
+
+	@Override
+	public void wrongGuess() {
+		lblStatus.setText("Sorry, Wrong Guess");
+		drawHangman(game.getHangmanStage());
```

Comment:
I think calling the `drawHangman` method would come under game logic, so the UI shouldn't implicitly call it.  This would be avoided by removing the GUI's `game` field